### PR TITLE
Deploy gateway backend as sgx

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

We want to run Ten Gateway inside an SGX enclave so users don't need to trust us anymore and we can prove which code we are running and that we are running it inside a TEE.

### What changes were made as part of this PR
- New dockerfile added to run gateway inside an enclave
- Files added required to run app in SGX (entry.sh, enclave.json)
- Changed VM type to a type that supports SGX
- Fixes for converting some objects to string that caused the gateway to crash in SGX mode

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

**Comment by tudor-malene** [comment]
> ### Why this change is needed
> Please provide a description and a link to the underlying ticket
> 
> ### What changes were made as part of this PR
> Please provide a high level list of the changes made
> 
> ### PR checks pre-merging
> Please indicate below by ticking the checkbox that you have read and performed the required [PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)
> 
> * [ ]  PR checks reviewed and performed

don't forget the descriptions

File: .github/workflows/manual-deploy-obscuro-gateway.yml [hunk 1]
```diff
@@ -207,9 +207,11 @@ jobs:
...
             && cd /home/obscuro/go-obscuro/ \
             && docker run -d -p 80:80 -p 81:81 --name "${{ env.VM_NAME }}" \
+              --entrypoint "/home/ten/go-ten/tools/walletextension/main/entry.sh" \
               -e OBSCURO_GATEWAY_VERSION="${{ GITHUB.RUN_NUMBER }}-${{ GITHUB.SHA }}" \
-               --log-opt max-file=3 --log-opt max-size=10m \
-               ${{ env.DOCKER_BUILD_TAG_GATEWAY }} \
-               -host=0.0.0.0 -port=80 -portWS=81 -nodeHost=${{ env.L2_RPC_URL_VALIDATOR }} -verbose=true \
-               -logPath=sys_out -dbType=mariaDB -dbConnectionURL="obscurouser:${{ secrets.OBSCURO_GATEWAY_MARIADB_USER_PWD }}@tcp(obscurogateway-mariadb-${{  github.event.inputs.testnet_type }}.uksouth.cloudapp.azure.com:3306)/ogdb" \
-               -rateLimitUserComputeTime=${{ env.GATEWAY_RATE_LIMIT_USER_COMPUTE_TIME }} -rateLimitWindow=${{ env.GATEWAY_RATE_LIMIT_WINDOW }} -maxConcurrentRequestsPerUser=${{ env.GATEWAY_MAX_CONCURRENT_REQUESTS_PER_USER }} '
+              -e OE_SIMULATION=1 \
```
Comment (tudor-malene) [inline .github/workflows/manual-deploy-obscuro-gateway.yml:212]: we need to be able to switch from simulation ON (for the local container) to simulation OFF ( for testnet)

File: tools/walletextension/enclave.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Build Stages:
+# build-base = downloads modules and prepares the directory for compilation. Based on the ego-dev image
+# build-enclave = copies over the actual source code of the project and builds it using a compiler cache
+# deploy = copies over only the enclave executable without the source
+#          in a lightweight base image specialized for deployment and prepares the /data/ folder.
+
+# Final container folder structure:
```
Comment (tudor-malene) [inline tools/walletextension/enclave.Dockerfile:7]: this comment here is not accurate

**Review by tudor-malene — COMMENTED** [review]
lgtm.
Few minor things

File: .github/workflows/manual-deploy-obscuro-gateway.yml (hunk 1 above)
Comment (zkokelj) [reply .github/workflows/manual-deploy-obscuro-gateway.yml]: We can do that now. This is github action file which is executed on Azure only.
I guess we will deploy only with non simulation mode, but if you think we need to deploy on Azure also with simulation mode sometimes, then I can introduce new env variable which can control that.

To run it locally we can simply run it like that:
```sh
docker run -d \
  -p 3000:3000 \
  -p 3001:3001 \
  --name local-gateway \
  --ulimit core=-1 \
  -v /tmp/cores:/tmp/cores \
  --security-opt seccomp=unconfined \
  --entrypoint "/home/ten/go-ten/tools/walletextension/main/entry.sh" \
  -e OBSCURO_GATEWAY_VERSION="local" \
  -e OE_SIMULATION=1 \
  gateway \
  ego run /home/ten/go-ten/tools/walletextension/main/main \
  -host=0.0.0.0 -port=3000 -verbose=true
```

**Pull request merged**